src/sentry/tasks/derive_code_mappings.py

```diff
@@ -17,42 +24,114 @@
+
+        integration = None
+        try:
+            integration = Integration.objects.filter(
+                organizations=organization.id,
+                provider="github",
+            )
+        except Integration.DoesNotExist:
+            logger.exception(f"Github integration not found for {organization.id}")
+
+        if integration is None or not integration.exists():
+            continue
+
+        integration = integration.first()
+        organization_integration = OrganizationIntegration.objects.filter(
+            organization=organization, integration=integration
+        )
+        if not organization_integration.exists():
+            continue
+
+        organization_integration = organization_integration.first()
+        install = integration.get_installation(organization.id)
+        trees: JSONData = install.get_trees_for_org()
+        for project, stacktrace_paths in project_stacktrace_paths.items():
+            code_mappings = derive_code_mappings(stacktrace_paths, trees)
+            set_project_codemappings(code_mappings, organization, organization_integration, project)
+
+
+# Given a list of code mappings, create a new repository project path
+# config for each mapping.
+def set_project_codemappings(
+    code_mappings: List[CodeMapping],
+    organization: Organization,
+    organization_integration: OrganizationIntegration,
+    project: Project,
+) -> None:
+    for code_mapping in code_mappings:
+        serializer = RepositoryProjectPathConfigSerializer(
+            data={
+                "project_id": project.id,
+                "stack_root": code_mapping.stacktrace_root,
+                "source_root": code_mapping.source_path,
+                "repo_id": code_mapping.repo,
+            },
+            context={
+                "organization": organization,
+                "organization_integration": organization_integration,
+            },
+        )
+        if serializer.is_valid():
+            serializer.save()
+        else:
+            logger.error(
+                "Error saving code mapping for organization %s: %s",
+                organization.id,
+                serializer.errors,
+            )
+
+
 @instrumented_task(  # type: ignore
     name="sentry.tasks.derive_code_mappings.identify_stacktrace_paths",
     queue="derive_code_mappings",
     max_retries=0,  # if we don't backfill it this time, we'll get it the next time
 )
 def identify_stacktrace_paths(
-    organizations: Optional[List[Organization]] = None,
-) -> Mapping[str, Mapping[str, List[str]]]:
+    organization: Organization, dry_run=False
+) -> Mapping[Project, List[str]]:
     """
-    Generate a map of projects to stacktrace paths for specified organizations,
-    or all active organizations if unspecified.
+    Generate a map of projects to stacktrace paths for an organization.
 
     This filters out non-python projects, or projects without an event
     in the last 7 days.
     """
-    if organizations is None:
```

Comment:
Moving this logic into `derive_code_mappings`, so identify_stacktrace_paths only handles one organization at a time.